src/main/java/tel/schich/javacan/BcmCanChannel.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * The MIT License
+ * Copyright © 2018 Phillip Schichtel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package tel.schich.javacan;
+
+import java.io.IOException;
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.channels.NotYetBoundException;
+import java.nio.channels.spi.SelectorProvider;
+
+import tel.schich.javacan.linux.LinuxNetworkDevice;
+
+/**
+ * The BcmCanChannel provides a wrapper around the CAN Broadcast Manager.
+ */
+public class BcmCanChannel extends AbstractCanChannel {
+    /**
+     * The MTU is calculated according the CAN documentation and assumes the use of FD frames. This way
+     * the buffer is large enough (~ 18 kB) for all use cases.
+     *
+     * @see https://www.kernel.org/doc/html/latest/networking/can.html#broadcast-manager-message-sequence-transmission
+     */
+    public static final int MTU = BcmMessage.HEADER_LENGTH + 256 * RawCanChannel.FD_MTU;
```

Comment:
the magic 256, while explained in the referenced documentation, should probably be extracted out into a constant on its own like `MAX_FRAMES_PER_MESSAGE`.

Reply:
right.